### PR TITLE
RND-462 Add `Masthead` component

### PR DESCRIPTION
**Pull request by kubama**
## Description
Added `Masthead` component

## Screenshots / Videos
image

## Checklist

- [X] My code follows the [UI Code Style](https://cloudifysource.atlassian.net/l/c/x8fq902p).
- [x] I followed the [UI Testing Guidelines](https://cloudifysource.atlassian.net/l/cp/udCVfvrx) and verified that all tests/checks have passed.
- [X] I verified if that change requires any update in the [documentation](https://cloudifysource.atlassian.net/l/c/4XKtPocy).
- [X] I added proper labels to this PR.

## Tests
Added stories, snapshots, loki images.

 

## Documentation
Added JSDocs

### CODE REVIEW

File: src/components/dds/Masthead/Masthead.tsx [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+
+export interface MastheadProps extends React.HTMLAttributes<HTMLDivElement> {
+    /**
+     * Content to be placed on the left side of the Dell logo
+     */
+    rightContent?: ReactNode;
+    /**
+     * Content to be placed on the right side of the Dell logo
+     */
+    leftContent?: ReactNode;
```
Comment (qooban) [inline src/components/dds/Masthead/Masthead.tsx:50]: I guess the description and prop name are mixed, am I correct?

**Comment by kubama** [comment]
I just noticed the svgs I included are part of the icon set and could be reused from there. I'll update this PR.

File: src/components/dds/Masthead/Masthead.tsx [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+    position: relative;
+`;
+
+const LogoContainer = styled.div<{ inverted: boolean; logoOrientation: LogoOrientation }>`
+    display: inline-block;
+    vertical-align: top;
+    img {
+        vertical-align: top;
+        filter: ${props => (props.inverted ? 'brightness(100)' : 'none')};
+        display: ${props => (props.logoOrientation === 'horizontal' ? 'inlilne' : 'block')};
```
Comment (qooban) [inline src/components/dds/Masthead/Masthead.tsx:25]:
```suggestion
        display: ${props => (props.logoOrientation === 'horizontal' ? 'inline' : 'block')};
```

File: src/components/dds/Masthead/Masthead.stories.tsx [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+export const invertedColors: MastheadStory = () => <Masthead inverted />;
+
+export const verticalLogo: MastheadStory = () => <Masthead inverted logoOrientation="vertical" />;
+
+export const additionalContent: MastheadStory = () => (
+    <Masthead
+        inverted
+        leftContent={<Icon name="menu-closed" style={{ color: 'white' }} />}
+        rightContent={<ActionMenu placement="end" style={{ position: 'absolute', right: 7 }} text="User menu" />}
```
Comment (qooban) [inline src/components/dds/Masthead/Masthead.stories.tsx:27]: Shouldn't we push right-content automatically (in `Masthead` component) to the right side? 
(maybe by having something similar to `LeftContentWrapper`)

File: src/components/dds/Masthead/Masthead.tsx [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+    }
+`;
+
+const LeftContentWrapper = styled.div`
+    margin: 10px;
```
Comment (qooban) [inline src/components/dds/Masthead/Masthead.tsx:30]: I suspect we should be using margin sizes like described in DDS:

> The 4-pixel grid is the Dell Design System standard. The recommended space between every element should be a multiple of 4 pixels, horizontally and vertically. 

In general: utility functions in `common/spacing.ts` can potentially help. We can add more there to provide support for working with CSS attributes directly. 

See
https://www.delldesignsystem.com/foundations/grid/#baseline-grid
https://www.delldesignsystem.com/foundations/grid/#vertical-spacing

**Comment by qooban** [comment]
I see nothing about `HeaderBar` component update in the description. 

Are you planning to do it in another PR or you decided not to do that?

**Comment by kubama** [comment]
> I see nothing about `HeaderBar` component update in the description.
> 
> Are you planning to do it in another PR or you decided not to do that?

Modifying `HeaderBar` would break backward compatibility of the whole package, I don't think this is something desired at this point. Creating a new component that is intended to replace the old one is a less troublesome path.

File: src/components/dds/Masthead/Masthead.stories.tsx (hunk 3 above)
Comment (kubama) [reply src/components/dds/Masthead/Masthead.stories.tsx:27]: We may put different things there, not necessarily just right-aligned dropdown menus. For example:
image
So for composer we will have additional 'Composer' title that should be actually left-aligned.

File: src/components/dds/Masthead/Masthead.tsx (hunk 2 above)
Comment (kubama) [reply src/components/dds/Masthead/Masthead.tsx]: Resolving as the code was removed

File: src/components/dds/Masthead/Masthead.tsx (hunk 1 above)
Comment (kubama) [reply src/components/dds/Masthead/Masthead.tsx]: Yes, fixed

File: src/components/dds/Masthead/Masthead.tsx (hunk 4 above)
Comment (kubama) [reply src/components/dds/Masthead/Masthead.tsx]: Applied

File: src/components/dds/Masthead/Masthead.stories.tsx (hunk 3 above)
Comment (qooban) [reply src/components/dds/Masthead/Masthead.stories.tsx:27]: OK.

**Review by qooban — APPROVED** [review]

**Comment by kubama** [comment]
@PatrycjaKubiczek The PR is ready for review

**Review by PatrycjaKubiczek — APPROVED** [review]

**Pull request merged**